pkgs/stdenv/generic/default.nix

```diff
@@ -75,6 +75,12 @@ let
     isUnfree (lib.lists.toList attrs.meta.license) &&
     !allowUnfreePredicate attrs;
 
+  allowInsecurePredicate = x: builtins.elem x.name (config.permittedInsecurePackages or []);
```

Comment:
Note that for `allowUnfreePredicate` the whole point is to allow user to define a more complicated policy (allow all `flashplayer` but nothing else). It is not described in the error message to avoid confusing users, though.

So I would prefer the same approach here:

```
  allowInsecurePredicate = x: ((config.allowInsecurePredicate or y: false) x) ||
    (builtins.elem x.name (config.permittedInsecurePackages or []));
```